core/src/main/java/org/apache/calcite/sql/SqlSessionTableFunction.java

```diff
@@ -42,25 +43,41 @@ public SqlSessionTableFunction() {
   private static class OperandMetadataImpl extends AbstractOperandMetadata {
     OperandMetadataImpl() {
       super(ImmutableList.of(PARAM_DATA, PARAM_TIMECOL, PARAM_KEY, PARAM_SIZE),
-          4);
+          3);
     }
 
     @Override public boolean checkOperandTypes(SqlCallBinding callBinding,
         boolean throwOnFailure) {
-      final SqlValidator validator = callBinding.getValidator();
-      if (!checkTableAndDescriptorOperands(callBinding, 2)) {
+      if (!checkTableAndDescriptorOperands(callBinding, 1)) {
+        return throwValidationSignatureErrorOrReturnFalse(callBinding, throwOnFailure);
+      }
+      if (!checkTimeColumnDescriptorOperand(callBinding, 1)) {
```

Comment:
In fact, as I recall `checkTableAndDescriptorOperands` will check whether the second argument is a descriptor, so this line might not be needed.

Reply:
Never mind. `checkTimeColumnDescriptorOperand` added a check for TIMESTAMP type for the one specified by DESCRIPTOR. We can keep this function here.